### PR TITLE
ParseIDMap should not ignore nonDigits

### PR DESCRIPTION
**Pull request by rhatdan**
This leads to people accidently putting characters into options
and not getting back good errors.

Fixes: https://github.com/containers/storage/issues/739

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Lint isn't happy with you @rhatdan

**Comment by rhatdan** [comment]
@nalind @saschagrunert @vrothberg @TomSweeneyRedHat @mtrmac @giuseppe PTAL

File: pkg/idtools/parser.go [hunk 1]
```diff
@@ -33,9 +26,12 @@ func parseTriple(spec []string) (container, host, size uint32, err error) {
 
 // ParseIDMap parses idmap triples from string.
 func ParseIDMap(mapSpec []string, mapSetting string) (idmap []IDMap, err error) {
-	stdErr := fmt.Errorf("error initializing ID mappings: %s setting is malformed", mapSetting)
+	stdErr := fmt.Errorf("error initializing ID mappings: %s setting is malformed: %q", mapSetting, mapSpec)
```
Comment (vrothberg) [inline pkg/idtools/parser.go:29]: I think the error should include the syntax. If I misconfigured it, I may not know what it should look like.

**Review by vrothberg — COMMENTED** [review]
I wanted to suggest to extend the tests but I don't spot unit tests for it.

I think parsers should have exhaustive unit tests. It's so easy to miss a corner case.

**Review by vrothberg — COMMENTED** [review]
LGTM

**Comment by rhatdan** [comment]
@nalind @TomSweeneyRedHat @saschagrunert @giuseppe PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by giuseppe** [comment]
could you rebase?

**Comment by TomSweeneyRedHat** [comment]
LGTM
as @giuseppe pointed out, needs a rebase

**Review by saschagrunert — APPROVED** [review]
LGTM :+1:
